### ISSUE TITLE
Object equivalency fails for Anonymous object Vs Private objects

### ISSUE THREAD

**Issue by SupunS** [ballerina-platform/ballerina-lang#10480]
```ballerina
type Employee object {
    int age;
};

public function main() {
    // semantic error
    object { int age; } p = new Employee();
}
```
Give the following semantic error:
```
incompatible types: expected '$anonType$0', found 'Employee'
```

**Comment by pubudu91** [comment]
Related issue: #14906

**Comment by pubudu91** [comment]
The above was fixed with PR #15180. Hence closing this.
